### DISCUSSION TITLE
Help Browser - Add new config

### DISCUSSION THREAD

**Discussion by flatcap** [neomutt/neomutt#4011] [Development]
> The Help Browser is a new feature to bring online help to NeoMutt.
> The code is in the [devel/help](https://github.com/neomutt/neomutt/tree/devel/help) branch.
>
> It adds a new config variable: `$help_doc_dir` which should point to a copy of the [`test-doc`](https://github.com/neomutt/test-doc) repo.
> Help can be activated by pressing <kbd>Alt-H</kbd>

### Part 1

Add a new config variable `$help_index_format` to format the index list appropriately.

Tasks:
- Add config variable to `init.h`
- Add global variable to `globals.[ch]`
- Alter `index.c` to use the new global

### Part 2

Add a new config variable `$help_simple_search` to search the help.

When a user searches for a string, but doesn't use a NeoMutt pattern, the string is transformed using [`$simple_search`](https://neomutt.org/guide/reference#simple-search).
This isn't appropriate for the help.

Tasks:
- Add config variable to `init.h`
- Add global variable to `globals.[ch]`
- Alter `pattern.c` to use the new global

The final part is a little trickier.  The pattern function doesn't know if this is a 'help' mailbox.

There are a couple of possible solutions:
- Pass the 'simple_search' variable in as a parameter
- Expand the search using 'simple_search' before calling

**Comment by jindraj** [comment]
> There are a couple of possible solutions:
> 
> * Pass the 'simple_search' variable in as a parameter
> * Expand the search using 'simple_search' before calling

which of these options is preferred/more reusable?

**Comment by flatcap** [comment]
> Expand the search using 'simple_search' before calling

This is a 'better' solution.
This separates the 'config' part from the 'searching' part.
The pattern code just works with what it's given.

**Comment by jfariasf** [comment]
Do you still need help with this?

**Comment by flatcap** [comment]
@jfariasf Yes please!

If you haven't already, there are a couple of guides for building NeoMutt:
- https://neomutt.org/distro.html (Quality varies)
- https://neomutt.org/dev/build

Then have a look through the code and see if the instructions, above, make sense.
And **ask lots of questions** -- either here, or on irc: `#neomutt` on `irc.freenode.net`
